content/docs/guides/continuous-delivery/troubleshooting-guide.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Other
+meta_desc: This page walks-through the common failures encountered while running Pulumi in CI/CD, as well as tips on how to fix them.
+menu:
+    userguides:
+        parent: cont_delivery
+        weight: 2
+---
+
+In order to understand the errors encountered during an automated pipeline execution, it is important to understand
+the steps involved in a typical CI/CD configuration for Pulumi regardless of the CI/CD service you are using.
+The type of failure you experience is likely related to one of these steps.
+
+## Overall Requirements
+
+In order to run a Pulumi command, the following are the requirements for any configuration involving Pulumi:
+
+* A Pulumi Access Token for the account you wish to use.
+Create one [here](https://app.pulumi.com/account/tokens) by logging in with the appropriate account.
+* A stack that you would like to update the automated pipeline.
+* Build tools (more on this below) based on the runtime of your Pulumi app.
+* Restore all dependencies for your Pulumi app.
+* The right cloud provider credentials.
+
+## Pulumi Access Token
+
+Pulumi has the smarts to know when it is run inside of a CI/CD service by detecting the environment configuration. When this happens,
+Pulumi automatically executes a `pulumi login` command using the non-interactive mechanism.
+
+### Tips
+
+* Ensure that the access token is saved in an environment variable called `PULUMI_ACCESS_TOKEN`.
+  * Most services support marking environment variables as a secret. Your Pulumi Access Token is indeed a sensitive value
+  and as such should be saved as a secret env var.
+* Ensure that the environment variable is actually accessible to the job that is running the `pulumi` command.
+* Several CI/CD systems have the concept of restricting access to environment secrets to specific branches. So make sure that your branch and the job
+running the `pulumi` command can access the env var.
+
+## Stack Name
+
+> Learn about [stacks]({{< relref "/docs/intro/concepts/stack" >}}) and their [configuration]({{< relref "/docs/intro/concepts/config" >}}).
+
+A stack represents a specific configuration state for your infrastructure resources. For a typical CI/CD pipeline, the stack must have been created
+beforehand using the `pulumi stack init` command and in the **appropriate organization**.
+
+### Tips
+
+* Ensure that the account represented by the token you are using, has access to the stack.
+  * This can lead to 404s being returned from the Pulumi Service because the token is invalid for any number of reasons.
+* Ensure that you use the fully-qualified stack name when passing the stack name to `pulumi` commands.
```

Comment:
perhaps an example  of a fqsn here?

Reply:
Oh shoot! I totally meant to do that. Glad you caught this. :)